assets/components/openshift-router/cluster-role-binding.yaml

```diff
@@ -9,5 +9,5 @@ subjects:
   namespace: openshift-ingress
 roleRef:
   kind: ClusterRole
-  name: openshift-ingress-router
```

Comment:
This file and cluster-role.yaml come in from the component via the rebase, so these changes are going to be overwritten on the next rebase, right? See https://github.com/openshift/microshift/blob/main/scripts/auto-rebase/assets.yaml#L24-L38

Why do we need to rename the roles?

Reply:
Oh, you are right.
OpenShift has the `system:router` clusterrole, which has almost the same permissions as this one and all tests are based on it given its "system" nature. Whenever they create a new router they bind to it, instead of the openshift-ingress-router. In OCP this is created by the openshift-apiserver bootstrap process.
As MicroShift does not have the OAS, should we include it in the bootstrap as a separate entity? There is nothing to rebase to.
https://github.com/openshift/openshift-apiserver/blob/master/pkg/bootstrappolicy/policy.go#L580-L593

Reply:
Does anything else in MicroShift (or running on MicroShift) need the role, or would it only be used for tests?

What's the difference between the 2 roles? 

Why do the tests expect `system:router` if it's not necessary for running the actual router code?

Reply:
All the tests are creating their own internal router to perform all the checks. These rely on the `system:router` cluster role, which is bound to the test namespace service account at the beginning.
The role we have in MicroShift belongs to the cluster ingress, and its a bit wider. It grants additional permissions for listing&watching namespaces and use hostnetwork SCC.
If an application wants to deploy additional routers it would need to bind to the ingress role, which is maybe a bit too wide.

Reply:
What are the circumstances where an application might want to deploy more routers?

I'm trying to understand whether we should make the test (or test job) ensure that role exists or whether we have a gap in our API compatibility with OCP that we need to close.